#! /bin/sh /usr/share/dpatch/dpatch-run
## 18dbpagesize_binpower.dpatch by  <hesso@pool.math.tu-berlin.de>
##
## DP: Make sure that the pagesize passed to db__set_pagesize() is
## DP: a power of two.

@DPATCH@
--- nvi-1.81.6.orig/common/exf.c	2009-03-09 01:48:01.695862889 +0100
+++ nvi-1.81.6/common/exf.c	2009-03-09 10:42:41.147866272 +0100
@@ -249,11 +249,10 @@
 		 * (vi should have good locality) or smaller than 1K.
 		 */
 		psize = ((sb.st_size / 15) + 1023) / 1024;
-		if (psize > 10)
-			psize = 10;
-		if (psize == 0)
-			psize = 1;
-		psize *= 1024;
+		if (psize >= 8) psize=8<<10;
+		else if (psize >= 4) psize=4<<10;
+		else if (psize >= 2) psize=2<<10;
+		else psize=1<<10;
 
 		F_SET(ep, F_DEVSET);
 		ep->mdev = sb.st_dev;
